injectproxy/enforce.go

```diff
@@ -111,13 +124,18 @@ func (ms Enforcer) EnforceNode(node parser.Node) error {
 	return nil
 }
 
-// EnforceMatchers replaces similar matcher present in
-// target label matchers or appends if not present.
-func (ms Enforcer) EnforceMatchers(targets []*labels.Matcher) []*labels.Matcher {
+// EnforceMatchers appends the configured label matcher if not present.
+// If the label matcher that is to be injected is present (by labelname) but
+// different (either by match type or value) an error is returned.
+func (ms Enforcer) EnforceMatchers(targets []*labels.Matcher) ([]*labels.Matcher, error) {
 	var res []*labels.Matcher
 
 	for _, target := range targets {
-		if _, ok := ms.labelMatchers[target.Name]; ok {
+		if matcher, ok := ms.labelMatchers[target.Name]; ok {
+			// matcher.String() returns something like "labelfoo=value"
+			if ms.errorOnReplace && matcher.String() != target.String() {
+				return res, ErrIllegalLabelMatcher
```

Comment:
do we want to tell which label is conflicting?

Reply:
Yes, good point. Added a commit to that effect.